src/main/java/org/opensearch/neuralsearch/query/HybridQueryBuilder.java

```diff
@@ -94,6 +96,30 @@ public HybridQueryBuilder add(QueryBuilder queryBuilder) {
         return this;
     }
 
+    /**
+     * Function to support filter on HybridQueryBuilder filter. Currently pushing down a filter
+     * to HybridQueryBuilder is not supported by design. We would simply check if the filter is valid
+     * and throw exception telling this is an unsupported operation. If the filter is null, then we do nothing and
+     * return.
+     * @param filter the filter parameter
+     * @return HybridQueryBuilder itself
+     */
+    public QueryBuilder filter(QueryBuilder filter) {
+        if (validateFilterParams(filter) == false) {
+            return this;
+        }
+        ListIterator<QueryBuilder> iterator = queries.listIterator();
+        while(iterator.hasNext()) {
+            QueryBuilder query = iterator.next();
+            if (query instanceof HybridQueryBuilder) {
```

Comment:
I don't think we need this check here?

Reply:
That will be handled by https://github.com/opensearch-project/neural-search/pull/498/files#diff-64ab60781cf1e9f7963a415406f49283b26945beaa46b7042e7cc5044d8d95b1R74

I don't think this method should worry about it.